app/src/main/java/com/es/findsoccerplayers/LoginActivity.java

```diff
@@ -0,0 +1,193 @@
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                GoogleSignInAccount account = task.getResult(ApiException.class);
+                firebaseAuthWithGoogle(account.getIdToken());
+                Toast.makeText(LoginActivity.this, "Loggato con successo", Toast.LENGTH_SHORT).show();
+            } catch (ApiException e) {
+                // Google Sign In failed, update UI appropriately
+                Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show();
+                prgBar.setVisibility(View.INVISIBLE);
+                // ...
+            }
+        }
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        prgBar.setVisibility(View.INVISIBLE);
+        Log.w("LoginAct","Act Creata");
+    }
+
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            FirebaseUser user = mAuth.getCurrentUser();
+                            updateUI(user);
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            prgBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(LoginActivity.this, (CharSequence) task.getException(), Toast.LENGTH_SHORT).show();
+                            updateUI(null);
+                        }
+                    }
+                });
+    }
+
+
+    private void signIn() {
+        prgBar.setVisibility(View.VISIBLE);
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    private void updateUI(FirebaseUser user) {
+        Intent intent = new Intent(LoginActivity.this,MainActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (acct==null){
+            if(backPressedTime + 2000 > System.currentTimeMillis()){
+                super.onBackPressed();
+                backToast.cancel();
+                finishAffinity();
+            }else{
+                backToast=Toast.makeText(getApplicationContext(), "Press back again to exit", Toast.LENGTH_SHORT);
+                backToast.show();
+            }
+            backPressedTime=System.currentTimeMillis();
+        }else{
+            startActivity(new Intent(getApplicationContext(),MainActivity.class));
```

Comment:
ma in che senso se l'utente preme back e `acct` è `null` mostri la MainActivity? La MainActivity deve essere mostrata soltanto dopo il login e solo se questo è successful.

Reply:
acct è un oggetto che rappresenta l'account loggato. se è null non c'è nessuno loggato. perciò se non c'è nessuno loggato e preme back 2 volte esce, mentre se è loggato torna alla main

Reply:
eh anche qui, io avevo in mente che la main activity venisse mostrata solo dopo il login, se va a buon fine